build/gen_stub.php

```diff
@@ -2760,40 +2760,36 @@ public function getClassSynopsisElement(DOMDocument $doc, array $classMap, itera
             "&InheritedProperties;"
         );
 
-        if (!empty($this->funcInfos)) {
-            $classSynopsis->appendChild(new DOMText("\n\n    "));
-            $classSynopsisInfo = $doc->createElement("classsynopsisinfo", "&Methods;");
-            $classSynopsisInfo->setAttribute("role", "comment");
-            $classSynopsis->appendChild($classSynopsisInfo);
+        $classSynopsis->appendChild(new DOMText("\n\n    "));
+        $classSynopsisInfo = $doc->createElement("classsynopsisinfo", "&Methods;");
+        $classSynopsisInfo->setAttribute("role", "comment");
+        $classSynopsis->appendChild($classSynopsisInfo);
 
-            $classReference = self::getClassSynopsisReference($this->name);
+        $classReference = self::getClassSynopsisReference($this->name);
 
-            if ($this->hasConstructor()) {
-                $classSynopsis->appendChild(new DOMText("\n    "));
-                $includeElement = $this->createIncludeElement(
-                    $doc,
-                    "xmlns(db=http://docbook.org/ns/docbook) xpointer(id('$classReference')/db:refentry/db:refsect1[@role='description']/descendant::db:constructorsynopsis[not(@role='procedural')])"
-                );
-                $classSynopsis->appendChild($includeElement);
-            }
+        $classSynopsis->appendChild(new DOMText("\n    "));
+        $includeElement = $this->createIncludeElement(
+            $doc,
+            "xmlns(db=http://docbook.org/ns/docbook) xpointer(id('$classReference')/db:refentry/db:refsect1[@role='description']/descendant::db:constructorsynopsis[not(@role='procedural')])"
+        );
```

Comment:
Does this create a fallback element that doesn't warn if it doesn't exist?

Reply:
Yes, it does:

https://github.com/php/php-src/blob/002d54db9f2d60a415ed4bd6112edf8466ee3c41/build/gen_stub.php#L2962-L2974